cpp/include/cudf/detail/copy_range.cuh

```diff
@@ -0,0 +1,305 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <utilities/bit_util.cuh>
+#include <utilities/cuda_utils.hpp>
+#include <utilities/column_utils.hpp>
+
+#include <cudf/copying.hpp>
+#include <cudf/types.hpp>
+#include <cudf/column/column_view.hpp>
+#include <cudf/utilities/error.hpp>
+#include <cudf/utilities/type_dispatcher.hpp>
+#include <rmm/device_scalar.hpp>
+#include <rmm/mr/device_memory_resource.hpp>
+
+#include <cub/cub.cuh>
+
+#include <cuda_runtime.h>
+
+#include <memory>
+
+namespace {
+
+// TODO: better define warp_size elsewhere (cuda_utils.hpp?).
+static constexpr cudf::size_type warp_size{32};
```

Comment:
I moved it to `utilities/detail/cuda.cuh` in https://github.com/rapidsai/cudf/pull/3263/files?file-filters%5B%5D=.cuh#diff-18cbb57e319d7642a4da9bf82c44004dR32